llm/yamls/hf_causal_lm/gpt2-small.yaml

```diff
@@ -0,0 +1,117 @@
+data_remote: &data_remote ./my-copy-c4
+data_local: &data_local ./my-copy-c4
+max_seq_len: &max_seq_len 1024
+tokenizer_name: &tokenizer_name gpt2
+
+# Run Name
+run_name: gpt2-small
+
+# Model
+model:
+  name: hf_causal_lm
+  hf_config: gpt2
```

Comment:
To get the HF model I've also been using just `hf_config: gpt2`, but are there any other config details we should consider adding???
(This is less of a review and more of an open question which should probably be ignored)

Reply:
I think the way it works is that `hf_config` can either be a string referencing a pretrained config living on `huggingface.co` or a local path to a fully defined HF Config. I think it's worth adding an example of that to the repo so people know it can be done.

Also, maybe the name for this param ought to be `config_name_or_path`, to make it more clear?

Reply:
hf_config is required here
https://github.com/mosaicml/benchmarks/blob/4d1a5a2b6e4934166371fcb6b136c2cb197e6532/llm/src/hf_causal_lm.py#L45

but i guess we can change that

Reply:
Since we're using huggingface transformer's AutoConfig.from_pretrained, maybe we should just use the same arg name they use [here](https://huggingface.co/docs/transformers/v4.24.0/en/model_doc/auto#transformers.AutoConfig.from_pretrained)
ie `AutoConfig.from_pretrained( pretrained_model_name_or_path, **kwargs )`